tests/phpunit/tests/media.php

```diff
@@ -5553,6 +5553,42 @@ public function test_wp_get_loading_optimization_attributes_if_loading_attr_pres
 		);
 	}
 
+	/**
+	 * @ticket 61436
+	 *
+	 * @covers ::wp_get_loading_optimization_attributes
+	 */
+	public function test_wp_get_loading_optimization_attributes_image_src_matched() {
+		add_filter(
+			'wp_get_loading_optimization_attributes',
+			static function ( $loading_attrs, $tag_name, $attr ) {
+				if (
+					'img' === $tag_name &&
+					isset( $attr['src'] ) &&
+					'https://example.org/a-specific-image.jpg' === $attr['src']
+				) {
+					$loading_attrs['fetchpriority'] = 'high';
```

Comment:
This is not properly testing the fix from this PR, because even today the image would already have `fetchpriority="high"` because it is the first image.

We need to use a value that the function wouldn't already set, to ensure that it's actually this filter that sets the value, for example `fetchpriority="low"`.
```suggestion
					$loading_attrs['fetchpriority'] = 'low';
```